### PR TITLE
Enhance has() API

### LINKED ISSUE #172: Unregistered has() features fail silently...
Three challenges with the `has()` API:
- If a feature is not registered, it just fails silently returning `undefined`.  This leads to potential logic errors, assuming a feature is being properly detected when in fact it isn't.  We should throw.
- Features are case sensitive, and while by convention, we keep labels all lower case, consumers of the API may unintentionally make mistakes... if we coerced the features to lowercase, it would avoid mistakes and errors.
- When `add` is passed with `override = false` it simply returns a `false` if the `add` fails because it is already present.  The way we currently consume the API, we don't handle those failures and it is likely that those downstream using the API will do the same.  We should consider throwing when the override fails, as it is clearly again something someone should be explicit about, instead of unintentional behaviour.

### PR DESCRIPTION
**Pull request by matt-gadd**
fixes #172

### CODE REVIEW

File: tests/unit/has.ts [hunk 1]
```diff
@@ -112,9 +126,21 @@ registerSuite({
...
+				try {
+					assert.isFalse(hasAdd('abc', false));
+				} catch (e) {
+					err = e;
+				}
+				assert.isTrue(err instanceof Error);
+				assert.equal(err.message, 'abc already exists and no overwrite flag was passed');
+			},
+
+			'feature name is lowercased'() {
```
Comment (agubler) [inline tests/unit/has.ts:141]: Do you also need a test that includes a test that when passing an uppercase string to `has` it lowercased?

Comment (agubler) [inline tests/unit/has.ts:141]: I can see it now, you can ignore.

Comment (matt-gadd) [inline tests/unit/has.ts:141]: thats covered above in the has test: https://github.com/dojo/core/pull/173/files#diff-9fb8a35085488b0cff49fd51446a500cR61

**Comment by kitsonk** [comment]
We should do this on `dojo/has` now.

**Pull request closed without merging**